test/helpers/cliCommandRunner.js

```diff
@@ -0,0 +1,49 @@
+// Import Node.js Dependencies
+import { fork } from "node:child_process";
+import { createInterface } from "node:readline";
+
+// Import Third-party Dependencies
+import { MockAgent, setGlobalDispatcher } from "undici";
+import stripAnsi from "strip-ansi";
+
+export async function runProcess(options) {
+  const { path, args = [], cwd = process.cwd(), undiciMockAgentOptions = null } = options;
+
+  const childProcess = fork(path, args, {
+    stdio: ["ignore", "pipe", "pipe", "ipc"],
+    cwd
+  });
+  // send needed options to trigger the `prepareProcess()` command.
+  childProcess.send(undiciMockAgentOptions);
+
+  const lines = [];
+
+  try {
+    const rStream = createInterface(childProcess.stdout);
+
+    for await (const line of rStream) {
+      lines.push(stripAnsi(line));
+    }
+  }
+  finally {
+    childProcess.kill();
+  }
+
+  return lines;
+}
+
+export function prepareProcess(command, args = process.argv.slice(2)) {
+  process.on("message", (undiciMockAgentOptions) => {
```

Comment:
.on or .once here? (I'm not sure by reading the method).

Reply:
Yes, `.once` is better since we don't expect multiple `runProcess`